Update package to google-genai in examples/Search_reranking_using_embeddings.ipynb

Jenniebn opened the pull request:
Update package to google-genai in `examples/Search_reranking_using_embeddings.ipynb`

- Updated google-generativeai to use google-genai package instead.
- Updated outputs and tested locally.
- Refactored imports
- Restructured the sections and updated instructions
- Formatted with nbfmt and nblint tool

Jenniebn committed:
Update package to google-genai in `examples/Search_reranking_using_embeddings.ipynb`

Jenniebn committed:
Update package to google-genai in examples/Search_reranking_using_embeddings.ipynb

Jenniebn committed:
format

Giom-V commented on examples/Search_reranking_using_embeddings.ipynb:
```diff
@@ -6,12 +6,12 @@
         "id": "Tce3stUlHN0L"
```
<pre class="ql-syntax" spellcheck="false">Line #4.    def to_markdown(text):
</pre><p>I don't think we need that function anymore. The output of the model should already be proper Markdown. Let's get rid of it and replace it by <code>display(Markdown())</code> everywhere. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Giom-V replied:
<pre class="ql-syntax" spellcheck="false">Line #42.            response = client.models.generate_content(model=model,
</pre><p>Ideally add a line break before <code>model</code> for readability. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Giom-V replied:
<p>I don't think this part is about Automatic function calling. I'd keep the old title.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Giom-V replied:
<p>Thanks for removing the temperature, I don't think it was bringing anything interesting here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Giom-V replied:
<p>I don't like this title and this part. My guess is that it was written when we didn't have SI, so I think we should change it to a system instruction (without the last line) and update the title and the text accordingly.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

Giom-V replied:
<p>Can you add some indentation to the string to improve readability? Like:</p>

<pre class="ql-syntax" spellcheck="false">instructions = """
    You have access to the Wikipedia API which you will be using
    to answer a user's query. Your job is to generate a list of search queries which
    
    [...]
    Here is the user's query: {query}
"""
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Giom-V replied:
<p>Since you removed the temperature setting, I think this sentence should be rephrased like "you will set" &gt; "you could set"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

Giom-V replied:
<p>I think this is where you should add a note explaining that it's a "Python SDK Only" feature.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Giom-V replied:
<pre class="ql-syntax" spellcheck="false">Line #2.        display(Markdown("###" + content.role + ":"))
</pre><p>You forgot a space after <code>###</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Giom-V replied:
<pre class="ql-syntax" spellcheck="false">Line #7.                print("Function call: {", part.function_call, "}")
</pre><p>I think it could be interesting to print <code>part.function_call.name, "(", part.function_call.args, ")"</code> or something like that to make it easier to read. Or maybe not so that we show what the output looks like. What you prefer. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Giom-V replied:
<pre class="ql-syntax" spellcheck="false">Line #2.    print(fc)
</pre><p>Can you better output the json, like it was before using <code>print(json.dumps(fc, indent=2))</code> ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Giom-V replied:
<pre class="ql-syntax" spellcheck="false">Line #3.    print(f"FunctionCall's name is {fc.name}")
</pre><p>I would remove this part, I don't see the value added</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>

Giom-V replied:
<pre class="ql-syntax" spellcheck="false">Line #3.        print(f"Function execution result: {summaries}")
</pre><p>I'm use <code>Markdown</code> to improve readability. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

Giom-V replied:
<p>Can you add <code># @title Helper function to embed the content</code> on the first line so we can collapse the code of the helper?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='50'/>

Giom-V replied:
<pre class="ql-syntax" spellcheck="false">Line #1.    res = client.models.generate_content(model=model,
</pre><p>Ideally add a line break before <code>model</code> for readability. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='62'/>

Giom-V replied:
<pre class="ql-syntax" spellcheck="false">Line #4.    about the said topic. Do not use factual information, instead use placeholders
</pre><p>Can you add some indentation to the string, a new line before the closing """, and before the closing parenthesis to improve readability?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='62'/>

Giom-V replied:
<p>Can you update the link to the cookbook? Somehow it works but I think it's better to call it "Cookbook" now and to point to the right url.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='70'/>

Giom-V commented on examples/Search_reranking_using_embeddings.ipynb:
```diff
@@ -6,12 +6,12 @@
         "id": "Tce3stUlHN0L"
```
<pre class="ql-syntax" spellcheck="false">Line #20.      # Note: newer models have more limited quota for API calls
</pre><p>What do you mean by that?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Jenniebn replied:
<p>Hi, I meant newer models have a more limited free quota for API calls compared to the older package. Please let me know if you prefer to delete this note</p>

Giom-V replied:
I see. If I'm not mistaken it's just because it's an experimental model. So I would maybe rephrase to "experimental models have more limited quota for API calls"

Giom-V commented on examples/Search_reranking_using_embeddings.ipynb:
```diff
@@ -6,12 +6,12 @@
         "id": "Tce3stUlHN0L"
```
<pre class="ql-syntax" spellcheck="false">Line #51.                                                            """),)
</pre><p>Nitpicking but I usually prefer to have some indentation within the string, so I would push the <code>"""</code> one step left to have it. I would also move the closing parenthesis of <code>generate_content</code> further left to align it with when it's opened. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Jenniebn replied:
<p>Feel free to let me know if the new format matches what you want</p>

Giom-V commented on examples/Search_reranking_using_embeddings.ipynb:
```diff
@@ -6,12 +6,12 @@
         "id": "Tce3stUlHN0L"
```
<p>If I'm not mistaken, with the new SDK the default value is now <code>true</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Jenniebn replied:
<p>I think according to <a href="https://github.com/googleapis/python-genai/blob/main/google/genai/types.py#L1557" target="_blank">this</a>, the defualt value is None, which will enable automatic function calling. I will change the instruction accordingly but please let me know if I understand anything wrong.</p><p> </p><p><code>class AutomaticFunctionCallingConfig(_common.BaseModel):</code></p><p><code>&nbsp;"""The configuration for automatic function calling."""</code></p><p> </p><p><code>&nbsp;disable: Optional[bool] = Field(</code></p><p><code>&nbsp;&nbsp;&nbsp;default=None,</code></p><p><code>&nbsp;&nbsp;&nbsp;description="""Whether to disable automatic function calling.</code></p><p><code>&nbsp;&nbsp;&nbsp;If not set or set to False, will enable automatic function calling.</code></p><p><code>&nbsp;&nbsp;&nbsp;If set to True, will disable automatic function calling.</code></p><p><code>&nbsp;&nbsp;&nbsp;""",</code></p><p><code>&nbsp;)</code></p><p> </p>

Giom-V commented:
Thanks for the submission @Jenniebn.

It's quite a complex notebook so that you for migrating it. 

I just added a bunch of comments to make the notebook easier to understand. Don't be alarmed by the number of comments, it's mostly minor stuff and some rewritting since the notebook has been written quite a long time ago (I guess on 1.0) .

And last thing, I think you already did, can you also make sure run all the cells to make sure everything works as expected?

Thanks again!

Giom-V commented:
@Jenniebn Ignore all suggestions from Gemini, it doesn't know anything about notebooks and any of the changes it recommends would break the notebook.

Jenniebn committed:
Merge branch 'google-gemini:main' into Search_reranking_using_embeddings

Jenniebn committed:
Merge branch 'google-gemini:main' into Search_reranking_using_embeddings

Jenniebn committed:
Temp implementation of comments (need final check and format)

Jenniebn committed:
Temp implementation of comments (need final check and format)

Jenniebn committed:
Temp implementation of comments (need final check and format)

Jenniebn committed:
Temp implementation of comments (need final check and format)

Jenniebn committed:
Temp implementation of comments (need final check and format)

Jenniebn committed:
Update Search_reranking_using_embeddings.ipynb

display error

Jenniebn committed:
Update Search_reranking_using_embeddings.ipynb

Jenniebn committed:
Merge branch 'google-gemini:main' into Search_reranking_using_embeddings

Jenniebn committed:
implemented all comments & format

Jenniebn commented on examples/Search_reranking_using_embeddings.ipynb:
```diff
@@ -11,7 +11,7 @@
     },
```
<pre class="ql-syntax" spellcheck="false">Line #5.    response = chat.send_message(query)
</pre><p>If I used <code>run all</code>  to run all the cells in the notebook, I could get <code>quota exceeded</code> error (just need to wait for a bit before rerun the cell), so it might be better to implement a error handling with retry here to handle that. Please let me know if you'd like that. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Giom-V replied:
<p>I'd say don't bother, unless you are really spamming, even a free account shouldn't be hitting the quota limits.</p>

Jenniebn commented on examples/Search_reranking_using_embeddings.ipynb:
```diff
@@ -11,7 +11,7 @@
     },
```
<pre class="ql-syntax" spellcheck="false">Line #3.        contents=f"""
</pre><p>I feel like this hypothetical document is not really effective in showing the user how reranking happens since the content generated by Gemini using its internal knowledge is mostly likely similar to <code>summaries</code> obtained using tools. I'm trying to think a better way to do this. Please feel free to let me know if you think this is necessary.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='61'/>

Jenniebn replied:
<p>I've changed the query to "Compare dog and cat characteristics." and show how the search results can be re-ranked based on user's own preference. Please feel free to let me know if you like this idea. If not, we can explore other options as well.</p>

Jenniebn commented:
> Thanks for the submission @Jenniebn.
> 
> It's quite a complex notebook so that you for migrating it.
> 
> I just added a bunch of comments to make the notebook easier to understand. Don't be alarmed by the number of comments, it's mostly minor stuff and some rewritting since the notebook has been written quite a long time ago (I guess on 1.0) .
> 
> And last thing, I think you already did, can you also make sure run all the cells to make sure everything works as expected?
> 
> Thanks again!

Thank you very much for your detailed review @Giom-V !

I've implemented all the comments, run and tested the code, and formatted it. I also left 2 comments. Could you please take a look and let me know what you think? Thank you in advance for your thoughts!

Giom-V commented on examples/Search_reranking_using_embeddings.ipynb:
```diff
@@ -11,7 +11,7 @@
     },
```
<pre class="ql-syntax" spellcheck="false">Line #5.    model=f"models/{MODEL_ID}"
</pre><p>You don't need to do that, just <code>MODEL_ID</code> should work all the time (if not let's file a bug) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Giom-V replied:
<pre class="ql-syntax" spellcheck="false">Line #45.                contents=textwrap.dedent(
</pre><p>Nit (only fix if there are other things to fix): I usually prefer to format long strings that way:</p>

<pre class="ql-syntax" spellcheck="false">&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; contents=textwrap.dedent(f"""
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;     Extract relevant information
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;     about user's query: {query}
    &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; From this source:

&nbsp; &nbsp;     &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; {page}

&nbsp; &nbsp; &nbsp; &nbsp;     &nbsp; &nbsp; &nbsp; &nbsp; Note: Do not summarize. Only Extract and return the relevant information
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; """,
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; ),
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Giom-V replied:
<p>I think you made a mistake when copy-pasting, there's no mittens here :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Giom-V commented on examples/Search_reranking_using_embeddings.ipynb:
```diff
@@ -11,7 +11,7 @@
     },
```
<pre class="ql-syntax" spellcheck="false">Line #1.    example = wikipedia_search(["What are LLMs?"])
</pre><p>Can you swap the prompt to "what is a transformer for LLMs" so that it does not cite a competitor in the output?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/google-gemini/cookbook/pull/637/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Jenniebn replied:
<p>The competitor still comes up occationally. Please let me know if you'd like a totally different query.</p>

Giom-V commented:
Thanks for all the updates @Jenniebn. I'm sorry if that feel like a lot of comments but I added a couple more remarks on the notebook. But it's shaping well, thanks for reworking that complex example.

When finished you'll also have to run the formatting script (cf. https://github.com/google-gemini/cookbook/actions/runs/14268185478/job/40090719430?pr=637). Ignore the lint issue, it's a know issue that you can ignore.

Jenniebn committed:
Merge branch 'google-gemini:main' into Search_reranking_using_embeddings

Jenniebn committed:
Merge branch 'google-gemini:main' into Search_reranking_using_embeddings

Jenniebn committed:
Update new query and re-ranking based on user's preference

Jenniebn committed:
Fixed notebook display error

Jenniebn committed:
Implemented new comments and new query

Jenniebn commented:
> Thanks for all the updates @Jenniebn. I'm sorry if that feel like a lot of comments but I added a couple more remarks on the notebook. But it's shaping well, thanks for reworking that complex example.
> 
> When finished you'll also have to run the formatting script (cf. https://github.com/google-gemini/cookbook/actions/runs/14268185478/job/40090719430?pr=637). Ignore the lint issue, it's a know issue that you can ignore.

Thank you for your detailed reviews again! I've implemented all your comments and also modified the section to better show how the search results can be re-ranked. Please feel free to let me know what you think, and I can modify accordingly.

Jenniebn committed:
Merge branch 'google-gemini:main' into Search_reranking_using_embeddings

Jenniebn committed:
Merge branch 'google-gemini:main' into Search_reranking_using_embeddings

Jenniebn committed:
update model id

Jenniebn commented on examples/Search_reranking_using_embeddings.ipynb:
```diff
@@ -11,7 +11,7 @@
     },
```
Hi Andy,

Thank you for your comment. Just updated the model id and also did the formatting. Feel free to let me know if anything else needs to change.

andycandy replied:
Nope, everything else lgtm!

Jenniebn commented:
@Giom-V could you please look at my updates and let me know if anything else needs editing? Thank you!
